utils/middlewares/hsl-jwt-validation-middleware.ts

```diff
@@ -0,0 +1,136 @@
+import * as E from "fp-ts/Either";
+import * as TE from "fp-ts/TaskEither";
+import { flow, pipe } from "fp-ts/lib/function";
+import * as jwt from "jsonwebtoken";
+import { getValidateJWT } from "@pagopa/ts-commons/lib/jwt_with_key_rotation";
+import { FiscalCode, NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { IRequestMiddleware } from "@pagopa/io-functions-commons/dist/src/utils/request_middleware";
+import {
+  getResponseErrorForbiddenNotAuthorized,
+  IResponseErrorForbiddenNotAuthorized,
+  IResponseSuccessJson,
+  ResponseErrorForbiddenNotAuthorized,
+  ResponseSuccessJson,
+  ResponseErrorInternal,
+  IResponseErrorInternal
+} from "@pagopa/ts-commons/lib/responses";
+
+import { readableReportSimplified } from "@pagopa/ts-commons/lib/reporters";
+import { AuthBearer } from "../../generated/definitions/external/AuthBearer";
+import { IConfig } from "../config";
+import { getHubSpidLoginClient } from "../../clients/hubSpidLogin";
+import { IntrospectSuccessResponse } from "../../generated/definitions/hub-spid-login/IntrospectSuccessResponse";
+
+export interface IHslJwtPayloadExtended extends jwt.JwtPayload {
+  readonly name: string;
+  readonly family_name: string;
+  readonly fiscal_number: FiscalCode;
+}
+
+type IjwtIntrospectionCall = (
+  token: NonEmptyString,
+  config: IConfig
+) => TE.TaskEither<
+  IResponseErrorInternal | IResponseErrorForbiddenNotAuthorized,
+  IResponseSuccessJson<IntrospectSuccessResponse>
+>;
+export const introspectionCall: IjwtIntrospectionCall = (token, config) =>
+  pipe(
+    TE.tryCatch(
+      () =>
+        getHubSpidLoginClient(
+          config.HUB_SPID_LOGIN_CLIENT_BASE_URL

```

Comment:
We should improve the client using an timeout abortable fetch.
You can see the implementation in [here](https://github.com/pagopa/io-backend/blob/5c594aeff63db9294ecf9a084655efcd2f5791b3/src/config.ts#L413)
The Application gateway has a timeout of 10 seconds, if the request take more than timeout time we can safely release the socket freeing resources.